server.js

```diff
@@ -0,0 +1,45 @@
+const { createServer } = require('node:http')
+const next = require('next')
+const { Server } = require('socket.io')
+
+const dev = process.env.NODE_ENV !== 'production'
+const hostname = 'localhost'
+const port = 3000
+// when using middleware `hostname` and `port` must be provided below
+const app = next({ dev, hostname, port })
+const handler = app.getRequestHandler()
+
+app.prepare().then(() => {
+	const httpServer = createServer(handler)
+
+	const io = new Server(httpServer, {
+		cors: {
+			origin: 'http://localhost:3000',
+		},
+	})
+
+	io.on('connection', (socket) => {
+		console.log('>>> server socket.id', socket.id)
+
+		socket.on('join-room', (roomId, userName) => {
```

Comment:
todo: Similar to what I said yesterday, let's move all the code related to the game logic (sockets) to its own folder and here in the server we simply call the methods :)

Reply:
For example, I can tell you have a console.log for every socket listener. By creating a custom function (a middleware), that function can do those common to every case without duplicating code, for example:

```js
function socketOn(name, cb) {
  socket.on(name, (...args) => {
    console.log('>>>', name', args)
    cb(...args);
  });
}
```

And then you call `socketOn` wherever needed, which will automatically do the `console.log` too:

```js
socketOn('client-message', (message, roomId) => {
  // do your stuff here...
})
```

Reply:
This will be worked on after we talk. I want to confirm what you are asking.

Reply:
It’s great!